pkg/phase.go

```diff
@@ -65,13 +66,27 @@ func (p *Phase) Exec(ctx context.Context, bp crv1alpha1.Blueprint, action string
 			if err = checkRequiredArgs(p.f.RequiredArgs(), args); err != nil {
 				return nil, errors.Wrapf(err, "Required args missing for function %s", p.f.Name())
 			}
+
+			if err = checkUnsupportedArgs(p.f.Arguments(), args); err != nil {
+				return nil, errors.Wrapf(err, "Checking supported args for function %s.", p.f.Name())
+			}
+
 			p.args = args
 		}
 	}
 	// Execute the function
 	return p.f.Exec(ctx, tp, p.args)
 }
 
+func checkUnsupportedArgs(supportedArgs []string, args map[string]interface{}) error {
```

Comment:
```suggestion
func checkSupportedArgs(supportedArgs []string, args map[string]interface{}) error {
```
[nit]

Reply:
we are actually checking if the argument that is provided in the function is not supported by that function. That is the reason I thought `checkUnSupportedArgs` was better. I will make the change for now and can decide later.